src/shared/modules/auth/index.js

```diff
@@ -0,0 +1,242 @@
+import {
+  getCredentialsFromAuthResult,
+  getInitialisationParameters,
+  getSSOProviderByIdpId,
+  temporarlyStoreUrlSearchParams
+} from './common'
+import {
+  AUTH_STORAGE_CODE_VERIFIER,
+  AUTH_STORAGE_STATE,
+  BEARER,
+  IMPLICIT,
+  PKCE
+} from './constants'
+import {
+  createStateForRequest,
+  createNonce,
+  createCodeVerifier,
+  createCodeChallenge,
+  authLog,
+  authDebug,
+  removeSearchParamsInBrowserHistory
+} from './helpers'
+import {
+  defaultCodeChallengeMethod,
+  defaultGrantType,
+  searchParamsToRemoveAfterAuthRedirect
+} from './settings'
+
+export const authRequestForSSO = idpId => {
+  const selectedSSOProvider = getSSOProviderByIdpId(idpId)
+  if (!selectedSSOProvider) {
+    const error = `Invalid OAuth2 endpoint: "${oauth2Endpoint}"`
+    authLog(error)
+    return error
+  }
+
+  temporarlyStoreUrlSearchParams()
+
+  const oauth2Endpoint = selectedSSOProvider.auth_endpoint
+  if (!oauth2Endpoint) {
+    const error = `Invalid OAuth2 endpoint: "${oauth2Endpoint}"`
+    authLog(error)
+    return error
+  }
+  authLog(`Using OAuth2 endpoint: "${oauth2Endpoint}" for idp_id: ${idpId}`)
+
+  const form = document.createElement('form')
+  form.setAttribute('method', 'GET')
+  form.setAttribute('action', oauth2Endpoint)
+
+  const ssoParams = selectedSSOProvider.params || {}
+  const state = createStateForRequest()
+  let params = {
+    ...ssoParams,
+    state
+  }
+  window.sessionStorage.setItem(AUTH_STORAGE_STATE, state)
+
+  const ssoExtraAuthParams = selectedSSOProvider.auth_params || {}
+  if (ssoExtraAuthParams) {
+    params = {
+      ...params,
+      ...ssoExtraAuthParams
+    }
+  }
+
+  authLog(
+    `Using the following authorization parameter: ${JSON.stringify(ssoParams)}`
+  )
+
+  const ssoConfig = selectedSSOProvider.config || {}
+  if (ssoConfig.implicit_flow_requires_nonce) {
+    params = {
+      ...params,
+      nonce: createNonce()
+    }
+    authLog(`Using nonce in authorization request`)
+  }
+
+  const _submitForm = (form, params) => {
+    for (const param in params) {
+      const input = document.createElement('input')
+      input.setAttribute('type', 'hidden')
+      input.setAttribute('name', param)
+      input.setAttribute('value', params[param])
+      form.appendChild(input)
+    }
+
+    document.body.appendChild(form)
+    form.submit()
+  }
+
+  if (selectedSSOProvider.auth_flow === PKCE) {
+    const codeChallengeMethod =
+      ssoConfig.code_challenge_method || defaultCodeChallengeMethod
+    authLog(
+      `Auth flow "PKCE", using code_challenge_method: "${codeChallengeMethod}"`
+    )
+
+    const codeVerifier = createCodeVerifier(codeChallengeMethod)
```

Comment:
This function can throw, but there's no catch...

Reply:
The parent should catch, rethrowing just to be explicit